src/views/radio.rs

```diff
@@ -120,15 +131,25 @@ impl<T> RadioButton<T> {
     }
 
     /// Selects this button, un-selecting any other in the same group.
-    pub fn select(&mut self) {
-        self.state.borrow_mut().selection = self.id;
+    pub fn select(&mut self) -> EventResult {
+        let mut state = self.state.borrow_mut();
+        state.selection = self.id;
+        if let Some(ref on_change) = state.on_change {
+            let on_change = Rc::clone(on_change);
+            let value = state.selection();
+            EventResult::with_cb(move |s| on_change(s, &value))
+        } else {
+            EventResult::Consumed(None)
+        }
     }
 
     /// Selects this button, un-selecting any other in the same group.
     ///
     /// Chainable variant.
     pub fn selected(self) -> Self {
-        self.with(Self::select)
+        self.with(|s| {
+            s.select();
```

Comment:
Can you add a comment indicating that we're dropping the potential callback here on purpose?